torch/nn/utils/clip_grad.py

```diff
@@ -52,8 +66,13 @@ def clip_grad_norm_(
     # avoids a `if clip_coef < 1:` conditional which can require a CPU <=> device synchronization
     # when the gradients do not reside in CPU memory.
     clip_coef_clamped = torch.clamp(clip_coef, max=1.0)
-    for g in grads:
-        g.detach().mul_(clip_coef_clamped.to(g.device))
+    if foreach:
+        clip_coef_clamped_scalar = clip_coef_clamped.item()
+        for (device, _), grads in per_device_and_dtype_grads.items():
```

Comment:
```suggestion
        for grads in per_device_and_dtype_grads.values():
```
if device isn't used in this for loop

Reply:
Thx, changed